### PR TITLE
Move async/await out of visitor function

### LINKED ISSUE #5: Does this work?
I tried it and it doesn't seem to be working for me and I think the reason is that remark plugins are not capable of being async. I'm pretty sure they're supposed to run synchronously. I cannot find any documentation about async plugins if that is possible. Am I just missing something?

### PR DESCRIPTION
**Pull request by isaacs**
Fix #5

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,16 +15,23 @@ const getBloquequote = async (url) => {
...
+      nodes.push([node, tweetLink]);
     }
-  });
+  })
+  for (let i = 0; i < nodes.length; i++) {
+    const nt = nodes[i];
+    const node = nt[0];
+    const tweetLink = nt[1];
+    console.log(`\n embeding tweet: ${tweetLink} \n`);
+    const embedData = await getBloquequote(tweetLink);
```
Comment (kentcdodds) [inline index.js]: Thanks @isaacs! I've built my own solution for this (needed more out-of scope features). In my solution I fire off all promises at the same time:

https://github.com/kentcdodds/kentcdodds.com/blob/037c9ac55ab59d1b7216a424b79024b857e626f4/plugins/remark-embedder/index.js#L8-L22

May be a good idea to use https://github.com/sindresorhus/p-throttle to limit it to 8 at once or something, but having these run at the same time will probably speed this up a lot.

File: index.js [hunk 2]
```diff
@@ -15,16 +15,23 @@ const getBloquequote = async (url) => {
...
-      node.value = embedData.html;
+      nodes.push([node, tweetLink]);
     }
-  });
+  })
+  for (let i = 0; i < nodes.length; i++) {
+    const nt = nodes[i];
+    const node = nt[0];
+    const tweetLink = nt[1];
+    console.log(`\n embeding tweet: ${tweetLink} \n`);
```
Comment (kentcdodds) [inline index.js]: FWIW, I found this console.log to be pretty annoying @jmolivas.

Comment (jmolivas) [reply index.js]: Agree like the idea may for `debug: true` @kentcdodds we should open a new issue to resport and have it fixed.

Comment (kentcdodds) [reply index.js]: I'm not planning on using this module as I have some specific needs for my site, so I'll leave this to you two :+1:

**Comment by jmolivas** [comment]
Fixed with PR https://github.com/weknowinc/gatsby-remark-twitter/pull/7

**Pull request closed without merging**